pandas/core/indexing.py

```diff
@@ -2018,6 +2018,8 @@ def _setitem_with_indexer_missing(self, indexer, value):
                 #  internals.concat logic
                 df = value.to_frame().T
                 df.index = [indexer]
+                # error: "List[Any]" has no attribute "name"
+                df.index.name = self.obj.index.name  # type: ignore[attr-defined]
```

Comment:
hmm this is actually modifying the indexer i think.

maybe
`df.index = pd.Index([indexer], name=self.obj.index.name)`?

Reply:
Good point, changed